src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -0,0 +1,3020 @@
+            }
+            fieldValues.remove(HOSPICE.REV_CNTR_NDC_QTY);
+            fieldValues.remove(HOSPICE.REV_CNTR_NDC_QTY_QLFR_CD);
+          } else if (lineItem.entry instanceof HealthRecord.Medication) {
+            HealthRecord.Medication med = (HealthRecord.Medication) lineItem.entry;
+            if (med.administration) {
+              hcpcsCode = "T1502";  // Administration of medication
+              fieldValues.put(HOSPICE.REV_CNTR_NDC_QTY, "1"); // 1 Unit
+              fieldValues.put(HOSPICE.REV_CNTR_NDC_QTY_QLFR_CD, "UN"); // Unit
+            }
+          }
+          if (hcpcsCode == null) {
+            continue;
+          }
+
+          fieldValues.put(HOSPICE.CLM_LINE_NUM, Integer.toString(claimLine++));
+          fieldValues.put(HOSPICE.REV_CNTR_DT, bb2DateFromTimestamp(lineItem.entry.start));
+          fieldValues.put(HOSPICE.HCPCS_CD, hcpcsCode);
+          fieldValues.put(HOSPICE.REV_CNTR_RATE_AMT,
+              String.format("%.2f", (lineItem.cost / Integer.max(1, days))));
+          fieldValues.put(HOSPICE.REV_CNTR_PMT_AMT_AMT,
+              String.format("%.2f", lineItem.coinsurance + lineItem.payer));
+          fieldValues.put(HOSPICE.REV_CNTR_TOT_CHRG_AMT,
+              String.format("%.2f", lineItem.cost));
+          fieldValues.put(HOSPICE.REV_CNTR_NCVRD_CHRG_AMT,
+              String.format("%.2f", lineItem.copay + lineItem.deductible + lineItem.pocket));
+          if (lineItem.pocket == 0 && lineItem.deductible == 0) {
+            // Not subject to deductible or coinsurance
+            fieldValues.put(HOSPICE.REV_CNTR_DDCTBL_COINSRNC_CD, "3");
+          } else if (lineItem.pocket > 0 && lineItem.deductible > 0) {
+            // Subject to deductible and coinsurance
+            fieldValues.put(HOSPICE.REV_CNTR_DDCTBL_COINSRNC_CD, "0");
+          } else if (lineItem.pocket == 0) {
+            // Not subject to deductible
+            fieldValues.put(HOSPICE.REV_CNTR_DDCTBL_COINSRNC_CD, "1");
+          } else {
+            // Not subject to coinsurance
+            fieldValues.put(HOSPICE.REV_CNTR_DDCTBL_COINSRNC_CD, "2");
+          }
+          rifWriters.writeValues(HOSPICE.class, fieldValues);
+        }
+
+        if (claimLine == 1) {
+          // If claimLine still equals 1, then no line items were successfully added.
+          // Add a single top-level entry.
+          fieldValues.put(HOSPICE.CLM_LINE_NUM, Integer.toString(claimLine));
+          fieldValues.put(HOSPICE.REV_CNTR_DT, bb2DateFromTimestamp(encounter.start));
+          fieldValues.put(HOSPICE.HCPCS_CD, "S9126"); // hospice per diem
+          rifWriters.writeValues(HOSPICE.class, fieldValues);
+        }
+      }
+    }
+  }
+
+  /**
+   * Export Home Health Agency visits for a single person.
+   * @param person the person to export
+   * @param stopTime end time of simulation
+   * @throws IOException if something goes wrong
+   */
+  private void exportSNF(Person person, long stopTime) throws IOException {
```

Comment:
I generated a sample 100 patient population, and in the output I had a snf.csv file that was totally empty. Is that the intended functionality? (ie, should the file be blank instead of just a header row?) I did confirm the file gets written to correctly if there are rows in it.

Reply:
Do you recall what params you used? In my testing I don't see a snf.csv file being created if there are no rows (vs creating an empty file).

Reply:
I tried it again a couple times and I haven't see any instances where the file is empty at the end of generation. Maybe I jumped the gun and looked at the file in the editor before generation was complete one time.

One other thing I do notice though is that if you generate one population where a file such as snf.csv is present, then a second population where nothing gets written to the file, the original file stays there. This is ok (the manifest doesn't refer to the file, and the same thing can happen with the CSV exporter if you change the included/excluded files) but it might be nice to document somehow. (Though I'm not sure where such a note would best fit)

Reply:
The BB RIF exporter is designed so that you can run Synthea multiple times to build up a population. We have a script that runs Synthea for each of the states and the resulting single sets of files then contain a national population.